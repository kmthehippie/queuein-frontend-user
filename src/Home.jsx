import { React, useState, useParams } from "react";
import Nav from "./components/Nav.jsx";

function Home() {
  const someFunction = () => {
    console.log("Hey we hit Sign Up Button!");
  };

  return (
    <div>
      <Nav />
      <div>
        <div>Write a bunch of good stuff about my app here!</div>
        <div>Some psycho actually liked my app! Here are the reviews!</div>
        <div>
          Some other additional info about my app! Like, it's free except for
          the database you are using!
        </div>
        <button type="button" onClick={someFunction}>
          Sign Up!
        </button>
      </div>
      <footer>
        <small>Some copyright secret small stuff here</small>
      </footer>
    </div>
  );
}

export default Home;
