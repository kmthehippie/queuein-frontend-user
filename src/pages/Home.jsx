import { React } from "react";
import { Link } from "react-router-dom";

function Home() {
  return (
    <div>
      <div>
        <div>Write a bunch of good stuff about my app here!</div>
        <div>Some psycho actually liked my app! Here are the reviews!</div>
        <div>
          Some other additional info about my app! Like, it's free except for
          the database you are using!
        </div>
        <Link to="/db/register">
          <button>Sign Up!</button>
        </Link>
      </div>
      <footer>
        <small>Some copyright secret small stuff here</small>
      </footer>
    </div>
  );
}

export default Home;
