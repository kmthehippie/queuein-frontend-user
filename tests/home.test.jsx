import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Home from "../src/Home";

//? Setup And Refreshing Testing

// describe("something truthy and falsy", () => {
//   it("true to be true", () => {
//     expect(true).toBe(true);
//   });

//   it("false to be false", () => {
//     expect(false).toBe(false);
//   });
// });

// App.test.jsx

// describe("Home component", () => {
//   it("renders Home", () => {
//     const { container } = render(<Home />);
//     expect(container).toMatchSnapshot();
//   });
//   it("renders radical rhinos after button click", async () => {
//     const user = userEvent.setup();

//     render(<Home />);
//     const button = screen.getByRole("button", { name: "Sign Up!" });

//     await user.click(button);

//     expect(screen.getByRole("heading").textContent).toMatch(/radical rhinos/i);
//   });
// });
