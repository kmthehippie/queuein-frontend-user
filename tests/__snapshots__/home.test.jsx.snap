// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > renders Home 1`] = `
<div>
  <div>
    <header>
      <div
        class="left"
      >
        <h1>
           
          Home
           
        </h1>
      </div>
      <div
        class="right"
      >
        <p>
          Login
        </p>
        <p>
          Sign Up
        </p>
      </div>
    </header>
    <div>
      <div>
        Write a bunch of good stuff about my app here!
      </div>
      <div>
        Some psycho actually liked my app! Here are the reviews!
      </div>
      <div>
        Some other additional info about my app! Like, it's free except for the database you are using!
      </div>
      <button
        type="button"
      >
        Sign Up!
      </button>
    </div>
    <footer>
      <small>
        Some copyright secret small stuff here
      </small>
    </footer>
  </div>
</div>
`;
